sljit_src/sljitNativeX86_common.c

```diff
@@ -414,41 +414,40 @@ static SLJIT_INLINE void sljit_unaligned_store_sw(void *addr, sljit_sw value)
 
 static void execute_cpu_id(sljit_u32 info[4])
 {
+#if (defined(SLJIT_CONFIG_X86_32) && SLJIT_CONFIG_X86_32) && defined(__GNUC__)
+	register int *linfo __asm__ ("esi");
```

Comment:
This is too much magic for me. Why using esi is a problem? We could use the same was as ebx: push/pop and not mentioning it in the clobbered list. Is is possible to use the stack directly in asm? Then we could allocate two local ariables on the stack and use `[esp/rsp+offset]` to save/restore ebp/esp.

Another alternative: since `info[1]` is never used, we could use ebx instead of esp, and pushes it twice on the stack.

Reply:
It is not a problem if used this way, the objective is that it does NOT use ebx, and both are possible since they are both valid on the ABI, the interesting part is that if they are not valid, then the compiler will save "esi" and restore automatically for us.

The problem with using it any other way, is that it can trigger bugs on the way the compiler allocates registers, since it is one of the ones it could chose to host the argument.

Reply:
We should reduce relying on the compiler, that is the main focus, if I understand correctly. Then be it minimal by manually saving/restoring everything by us, and tell nothing about it for the compiler.

Reply:
One way to do that is to rely more in the system provided calls (at least for Windows) and that is what the [mingw32 branch](https://github.com/carenas/sljit/tree/mingw32) was proposing.

When that is not possible, inlined assembler makes for the most optimal alternative IMHO.

Reply:
We agree on that. I just want to remove the gcc specific part from the picture, and make the inlined assembly rely on minimal things. I still remember the TinyC issues.

Reply:
implemented differently with  #251